### ISSUE TITLE
[salt-ssh] fails when targeting IP without a roster file

### ISSUE THREAD

**Issue by rossengeorgiev** [saltstack/salt#49968]
### Description of Issue/Question
I've discovered that you can now target hosts without a roster file. 
There was an issue with reverse DNS when using a host, which appears to have been fixed.

https://github.com/saltstack/salt/issues/49840 (PR: https://github.com/saltstack/salt/pull/48771)

I tried using an IP as the target and that fails as a reverse lookup is performed. Looking at the above PR, it doesn't look like this issue was addressed. I'm not sure if its intended for IPs to be supported as target. It seem reasonable for this to be possible. Arguably, salt-ssh shouldn't be doing reverse dns lookup for IPs as it may result the output having a hostname in the output, and mismatch what was set as the target. Below are the related debug lines:

```python
[DEBUG   ] salt.utils.network.ip_to_host(u'192.168.22.131') failed: [Errno 1] Unknown host
...
[DEBUG   ] Adding minions for job 20181010110305504311: [None]
```

### Setup
Just install latest salt-ssh

### Steps to Reproduce Issue
```python
[root@centos7-5 ~]# salt-ssh -l debug '192.168.22.131' test.ping

[DEBUG   ] Reading configuration from /etc/salt/master
[DEBUG   ] Using cached minion ID from /etc/salt/minion_id: centos7-5.local
[DEBUG   ] Configuration file path: /etc/salt/master
[WARNING ] Insecure logging configuration detected! Sensitive data may be logged.
[DEBUG   ] salt.utils.network.ip_to_host(u'192.168.22.131') failed: [Errno 1] Unknown host
[DEBUG   ] LazyLoaded flat.targets
[DEBUG   ] LazyLoaded jinja.render
[DEBUG   ] LazyLoaded yaml.render
[DEBUG   ] compile template: /etc/salt/roster
[DEBUG   ] Jinja search path: [u'/var/cache/salt/master/files/base']
[DEBUG   ] LazyLoaded roots.envs
[DEBUG   ] Could not LazyLoad roots.init: 'roots.init' is not available.
[DEBUG   ] Updating roots fileserver cache
[PROFILE ] Time (in seconds) to render '/etc/salt/roster' using 'jinja' renderer: 0.0428719520569
[DEBUG   ] Rendered data from file: /etc/salt/roster:
# Sample salt-ssh config file
#web1:
#  host: 192.168.42.1 # The IP addr or DNS hostname
#  user: fred         # Remote executions will be executed as user fred
#  passwd: foobarbaz  # The password to use for login, if omitted, keys are used
#  sudo: True         # Whether to sudo to root, not enabled by default
#web2:
#  host: 192.168.42.2


[DEBUG   ] Results of YAML rendering:
{}
[PROFILE ] Time (in seconds) to render '/etc/salt/roster' using 'yaml' renderer: 0.00113391876221
[DEBUG   ] Matched minions: {}
[DEBUG   ] salt.utils.network.ip_to_host(u'192.168.22.131') failed: [Errno 1] Unknown host
[DEBUG   ] LazyLoaded roots.envs
[DEBUG   ] Could not LazyLoad roots.init: 'roots.init' is not available.
[DEBUG   ] Updating roots fileserver cache
[DEBUG   ] LazyLoaded local_cache.prep_jid
[DEBUG   ] Adding minions for job 20181010110305504311: [None]
[ERROR   ] An un-handled exception from the multiprocessing process 'MultiprocessingProcess-1' was caught:
Traceback (most recent call last):
  File "/usr/lib/python2.7/site-packages/salt/utils/process.py", line 747, in _run
    return self._original_run()
  File "/usr/lib64/python2.7/multiprocessing/process.py", line 114, in run
    self._target(*self._args, **self._kwargs)
  File "/usr/lib/python2.7/site-packages/salt/client/ssh/__init__.py", line 518, in handle_routine
    **target)
  File "/usr/lib/python2.7/site-packages/salt/client/ssh/__init__.py", line 920, in __init__
    self.shell = salt.client.ssh.shell.gen_shell(opts, **args)
  File "/usr/lib/python2.7/site-packages/salt/client/ssh/shell.py", line 63, in gen_shell
    shell = Shell(opts, **kwargs)
  File "/usr/lib/python2.7/site-packages/salt/client/ssh/shell.py", line 90, in __init__
    self.host = host.strip('[]')
AttributeError: 'NoneType' object has no attribute 'strip'
Process MultiprocessingProcess-1:
Traceback (most recent call last):
...
```

### Versions Report
```yaml
Salt Version:
           Salt: 2018.3.2

Dependency Versions:
           cffi: 1.6.0
       cherrypy: Not Installed
       dateutil: 1.5
      docker-py: Not Installed
          gitdb: Not Installed
      gitpython: Not Installed
          ioflo: Not Installed
         Jinja2: 2.7.2
        libgit2: Not Installed
        libnacl: Not Installed
       M2Crypto: Not Installed
           Mako: Not Installed
   msgpack-pure: Not Installed
 msgpack-python: 0.5.6
   mysql-python: Not Installed
      pycparser: 2.14
       pycrypto: 2.6.1
   pycryptodome: Not Installed
         pygit2: Not Installed
         Python: 2.7.5 (default, Jul 13 2018, 13:06:57)
   python-gnupg: Not Installed
         PyYAML: 3.11
          PyZMQ: 15.3.0
           RAET: Not Installed
          smmap: Not Installed
        timelib: Not Installed
        Tornado: 4.2.1
            ZMQ: 4.1.4

System Versions:
           dist: centos 7.5.1804 Core
         locale: UTF-8
        machine: x86_64
        release: 3.10.0-862.11.6.el7.x86_64
         system: Linux
        version: CentOS Linux 7.5.1804 Core
```

**Comment by Ch3LL** [comment]
ping @gtmanfred seems you added this PR. mind taking a look here?

**Comment by gtmanfred** [comment]
That pr does not address the issue. this is just a duplicate of https://github.com/saltstack/salt/issues/49840

The PR just removes doing reverse dns lookups on hostnames.

Still waiting on feedback from the rest of team-ssh on that issue.

**Comment by rossengeorgiev** [comment]
I guess mentioning that issue has derailed this one a bit. Opss 🙂

**Comment by gtmanfred** [comment]
No, the issue is the same, it tries to resolve it addresses to reverse dns.

**Comment by Ch3LL** [comment]
okay i will close this as a duplicate and we can monitor the other issue. thanks for clarifying @gtmanfred
